crates/sui-framework/sources/collectible.move

```diff
@@ -0,0 +1,446 @@
+    /// implementation, it is possible to transfer objects TO `Collectible`s.
+    public fun uid_mut<T: store + drop>(_w: T, self: &mut Collectible<T>): &mut UID { &mut self.id }
+
+
+    /// Create a Collection and receive a `CollectionManagerCap`. This object
+    /// is similar to the `coin::TreasuryCap` in a way that it manages total
+    /// supply and grants the owner permission to mint and burn new `Collectible`s.
+    ///
+    /// It also holds information about the `total_supply` and prevents from
+    /// minting more than `max_supply`.
+    public fun create_collection<T: store + drop>(
+        _w: T, max_supply: Option<u64>, ctx: &mut TxContext
+    ): CollectionManagerCap<T> {
+        CollectionManagerCap {
+            max_supply,
+            id: object::new(ctx),
+            total_supply: 0,
+        }
+    }
+
+    /// Mint a new `Collectible` using `CollectionManagerCap`.
+    public fun mint<T: store + drop>(
+        cap: &mut CollectionManagerCap<T>, info: T, ctx: &mut TxContext
+    ): Collectible<T> {
+        cap.total_supply = cap.total_supply + 1;
+
+        // if `max_supply` is set -> enforce it
+        if (option::is_some(&cap.max_supply)) {
+            assert!(cap.total_supply <= *option::borrow(&cap.max_supply), EMaxSupplyReached);
+        };
+
+        Collectible {
+            info,
+            id: object::new(ctx),
+            edition: *&cap.max_supply,
+            unique_number: cap.total_supply,
+        }
+    }
+
+
+    /// Implementable `transfer::transfer` function.
+    public fun transfer<T: store + drop>(_w: T, nft: Collectible<T>, to: address) {
+        transfer::transfer(nft, to);
+    }
+
+    /// Implementable `transfer::transfer_to_object` function.
+    public fun transfer_to_object<T: store + drop, S: key + store>(_w: T, nft: Collectible<T>, obj: &mut S) {
+        transfer::transfer_to_object(nft, obj)
+    }
+
+    /// Implementable `transfer::transfer_to_object_id` function.
+    public fun transfer_to_object_id<T: store + drop>(_w: T, nft: Collectible<T>, owner_id: &mut UID) {
+        transfer::transfer_to_object_id(nft, owner_id)
+    }
+
+    /// Implementable `transfer::share_object` function.
+    public fun share_object<T: store + drop>(_w: T, nft: Collectible<T>) {
+        transfer::share_object(nft)
+    }
+}
+
+#[test_only]
+/// Defines a `Tricky` Collectible standard which has few important properties:
+///
+/// - Can be traded on multiple marketplaces at the same time;
+/// - Considered owned while listed on a marketplace;
+/// - This standard provides ownership management while keeping metadata definition
+/// up to the implementers;
+///
+///
+/// The way it is implemented is the following:
+/// - Every Collectible<Tricky> is a shared object with the `owner` field which
+/// implements logical ownership.
+/// - As long as user is the owner he is free to change ownership to any other address
```

Comment:
As long as a user is the `owner`, that user can change

changing "he" for inclusivity
"is free to" for localization